### PR TITLE
docs(guide): add dart version of structural directives

### PR DESCRIPTION
**Pull request by adaojunior**

### CODE REVIEW

**Comment by adaojunior** [comment]
I messed up in the other [pull request](https://github.com/angular/angular.io/pull/696)

**Comment by kwalrath** [comment]
It looks like you haven't incorporated all the comments from #696, so I'll wait for another commit before taking a final look.

**Comment by adaojunior** [comment]
@kwalrath I removed the unused file and changed to relative imports.

File: public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@Component(
+    selector: 'heavy-loader',
+    template: '<span>heavy loader #{{id}} on duty!</span>')
+class HeavyLoaderComponent implements OnInit, OnDestroy {
+  int id = nextId++;
+  @Input() List<String> logs;
+
+  // Mock todo: get 10,000 rows of data from the server
+  ngOnInit() => _log(
+      "heavy-loader ${id} initialized, loading 10,000 rows of data from the server");
```
Comment (zoechi) [inline public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart]: Any feedback on this string interpol style yet? 
The Dart style guide discourages `{}` here

Comment (adaojunior) [inline public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart]: @zoechi Aparently they're only necessary when it is a expression or you're calling a method in a object. I'll remove them. Thanks

**Comment by kwalrath** [comment]
@sanfordredlich could you please do a technical review? I'm especially curious about the `_tick() => new Future(() {})` code in heavy_loader_component.dart. (It's derived from the TS code, but using a Timer resulted in more complicated code, so I suggested new Future() as a way of kicking the event thread.)

File: public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  // Mock todo: clean-up
+  ngOnDestroy() => _log("heavy-loader $id destroyed, cleaning up");
+
+  _log(String msg) {
+    logs.add(msg);
+    _tick();
+  }
+
+  /// Triggers the next round of Angular change detection
+  /// after one turn of the JavaScript cycle
```
Comment (zoechi) [inline public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart]: Is "JavaScript cycle" a good explanation here?

Comment (kwalrath) [inline public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart]: Good question. How about "browser event loop"? We should perhaps make this change to the JS/TS samples, too.

Comment (zoechi) [inline public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart]: Sounds good and should fly in JS land as well

Comment (adaojunior) [inline public/docs/_examples/structural-directives/dart/lib/heavy_loader_component.dart]: done, also sent a PR to change it in the TS example.
https://github.com/angular/angular.io/pull/710

**Comment by kwalrath** [comment]
I'm going to merge this PR, so we can point to this code from a placeholder page. We'll do a final technical review before I write & publish the real page.

**Pull request closed without merging**